### DISCUSSION TITLE
Adicionar suporte a outros idiomas

### DISCUSSION THREAD

**Discussion by george-gca** [levxyca/diciotech#243] [Polls]
Discussão iniciada em #185, sendo que uma 1a PR foi criada em #140 e outra em #217.

Contexto: em #140 foi adicionada a opção de tradução dos cards, porém faltou de outras partes da interface. Em #217 a tradução completa da UI é feita, porém foi adicionada uma nova ferramenta: [Jekyll](https://jekyllrb.com/).

Vou listar aqui o que eu e a @levxyca vemos como pontos a se considerar:
1. Adicionar suporte a outro idioma deveria ser uma prioridade?
2. Quão difícil é e será para uma pessoa fazer uma contribuição de termos novos?
3. Quão difícil é e será para uma pessoa fazer uma contribuição que não seja termos novos?

### Adicionar suporte a outro idioma deveria ser uma prioridade?

Isso é uma decisão que pensamos que deve ser discutida com a comunidade. Enquanto sabemos que existe um projeto similar de língua inglesa ([Dev-Encyclopedia](https://github.com/Buzzpy/Dev-Encyclopedia)), não sabemos se existe em espanhol, por exemplo. Então adicionar outros idiomas pode ter um diferencial pra outras línguas.

### Quão difícil é e será para uma pessoa fazer uma contribuição de termos novos?

Acredito que não seja difícil atualmente, visto que os termos estão em ordem alfabética e é fácil identificar onde e como inserir um novo termo. Porém após inserirmos suporte ao [Prettier](https://prettier.io/) em #240 isso vai ficar um pouco mais chatinho, pois a pessoa precisa se atentar ao padrão usado (identação, se as chaves e colchetes abrem na mesma linha, etc).

Porém um ponto a se pensar: ao termos suporte a novos idiomas, quando uma pessoa usuária adicionar um novo termo em sua língua materna (ex: português), os outros idiomas vão ficar com verbetes a menos, e isso pode criar um débito técnico. Claro, sempre dá pra abrir issues indicando a falta desses termos traduzidos e alguém da comunidade inserir a tradução.

### Quão difícil é e será para uma pessoa fazer uma contribuição que não seja termos novos?

Atualmente contribuir com código novo não é tão complicado, porém depende. A pessoa deve se atentar, por exemplo, que as mudanças de visual (css) não devem ser feitas nos arquivos css, e sim sass. Isso já envolve instalar uma nova ferramenta localmente (sass), e essa parte ainda pode ser melhorada na documentação.

Ao inserirmos suporte a novos idiomas por meio do `Jekyll`, isso envolve criar toda uma documentação explicando como uma pessoa pode instalar o ambiente de desenvolvimento para testar as mudanças localmente. E para a instalação de fato, temos 2 opções:

1. instalar o Jekyll na máquina e consequentemente suas dependências. Isso envolveria mais uma ferramenta além do sass, apesar de que o sass pode ser substituído por um plugin do Jekyll como o [jekyll-minifier](https://github.com/digitalsparky/jekyll-minifier) ou o [jekyll-terser](https://github.com/RobertoJBeltran/jekyll-terser)
2. instalar o docker na máquina local, e então usar um [devcontainer](https://aka.ms/devcontainer.json). IDEs como o VSCode detectam automaticamente que o repositório tem um devcontainer quando aberto, e instala e configura todo o ambiente automaticamente.

Visto isso tudo, o que vocês acham?

**Comment by george-gca** [comment]
@levxyca qual é a melhor forma de tornar isso visível pra geral?

**Comment by levxyca** [comment]
@george-gca posso divulgar nas minhas redes como normalmente faço pra dar mais visibilidade, aqui dentro do próprio github acho que só se a pessoa vier aqui olhar mesmo, diretamente.

**Reply by george-gca** [reply]
Então, por favor, divulga. O que acha de darmos um prazo pra termos uma decisão?

**Reply by levxyca** [reply]
@george-gca consegui fazer as postagens hoje, podemos deixar aberto até a próxima sexta 8/11, independente de termos mais votos ou não, tomamos uma decisão a partir disso 😊

https://x.com/levxyca/status/1852378777923231844
https://bsky.app/profile/did:plc:b4ybm26q64u67bigzzznrqnj/post/3l7vjurwq362p
https://www.linkedin.com/feed/update/urn:li:activity:7258144603344621568/
